### ISSUE TITLE
"New Version Available" check

### ISSUE THREAD

**Issue by aniri** [code4romania/mon-vot-android-kotlin#189]
Every time a user opens the app we should check if they're on the latest version on the Play Store. If they're not, we should show a dialog that tells them to update.

Ideally we'd have 2 flags on RemoteConfig:
- `android_check_update_available`
- `android_force_update` - if this is `true`, we should not let the user get past that screen until she's on the latest version.

Both flags will default to `true`.

Should look similar to the iOS implementation:
![Simulator Screen Shot - iPhone 11 Pro - 2020-01-13 at 11 06 02](https://user-images.githubusercontent.com/1358055/72244342-c1ed4200-35f6-11ea-921a-45ff099050f9.png)

Texts: 
EN
- title: App Update Available
- text: A new app version is available. Please update the app to use the latest features

RO
- title: Actualizare disponibilă
- text: O nouă versiune a aplicației este disponibilă. Te rugăm să faci update pentru a folosi cele mai recente funcționalități.

**Comment by lukstbit** [comment]
Is the dialog UI an absolute requirement? 
If not we could use the default [in-app updates](https://developer.android.com/guide/playcore/in-app-updates). This would require a dependency on the _com.google.android.play:core_ library from Google.

**Comment by aniri** [comment]
> Is the dialog UI an absolute requirement?
> If not we could use the default [in-app updates](https://developer.android.com/guide/playcore/in-app-updates). This would require a dependency on the _com.google.android.play:core_ library from Google.

@lukstbit no, the dialog is not a requirement, it is just a suggestion. You are right, using the default in-app updates would be better :)

**Comment by aniri** [comment]
hey @lukstbit :smiley: ! would you be interested in handling this task with the default in-app updates?

**Comment by VladCuciureanu** [comment]
@aniri I'll take this issue if it's still free

**Comment by aniri** [comment]
> @aniri I'll take this issue if it's still free

@VladCuciureanu yep, it's free! thanks!

**Comment by pedrofsn** [comment]
@aniri this task is opened yet? If yes, I could help!

**Comment by aniri** [comment]
> @aniri this task is opened yet? If yes, I could help!

@pedrofsn thanks for your interest, but this task is already assigned to someone.

**Comment by VladCuciureanu** [comment]
@pedrofsn I would be willing to pass the issue over to you. I couldn't figure out how to implement and/or test this. Maybe you'll have more luck :)

**Comment by pedrofsn** [comment]
Hi, i did this task but it's getting some firebase erros. 
I've tried to initialize with and without koin but not worked. If it can be fixed, the update flow will works!

The error:
`Unable to start activity ComponentInfo{ro.code4.monitorizarevot/ro.code4.monitorizarevot.ui.login.LoginActivity}: java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process ro.code4.monitorizarevot. Make sure to call FirebaseApp.initializeApp(Context) first.`
[...]
`Make sure to call FirebaseApp.initializeApp(Context) first.`
